content/src/view.js

```diff
@@ -3172,6 +3380,147 @@ function setupHpanelBox(box) {
   });
 }
 
+function curvedVertical(x1, y1, x2, y2) {
+  var radius = Math.abs(y1 - y2);
+  var line = [];
+
+  return 'M'+ x1 + "," + y1 + " " + 'A'+ radius + "," + radius + " 1 0,1 " + x2 + "," + y2;
+}
+
+//$(window).resize(..) event
+
+function arrow(pane, arrows, traceEventNum, show_fade){
+  /* note: we expect arrow_lines to be an array of key value pairs where 
+  each key is a color for the arrow, and each value is a list of location pairs
+  to draw an arrow on.   */
+
+  var arrow_data = arrows[traceEventNum]; 
+  var startcoordsBefore = null;
+  var endcoordsBefore = null;
+  var offset_top_before =  0; 
+  var offset_left_before = 0; 
+  var firstBeforeLoc = {};
+  var secondBeforeLoc = {};
+  var firstAfterLoc = {};
+  var secondAfterLoc = {};
+  var startcoordsAfter = null;
+  var endcoordsAfter = null;
+  var offset_top_after =  0; 
+  var offset_left_after = 0; 
+  var before_arrowtext = "";
+  var after_arrowtext = "";
+
+  $(".arrow").remove();
+
+  var block_mode = null;
+  if (pane){
+    block_mode = true;
+    if(!getPaneEditorBlockMode(pane)){block_mode = false;}
+  }
+
+  if (arrow_data){
+    if(arrow_data["before"]){
+      firstBeforeLoc = arrow_data['before']["first"];
+      secondBeforeLoc = arrow_data['before']['second'];
+    }
+    if(arrow_data["after"]){
+      firstAfterLoc = arrow_data["after"]["first"];
+      secondAfterLoc = arrow_data["after"]["second"];
+    }
+  }
+          
+  if (firstBeforeLoc.first_line != undefined && secondBeforeLoc.first_line != undefined){
+    
+    if (block_mode){
+      var dropletEditor = state.pane[pane].dropletEditor;
+      var startBoundsBefore = dropletEditor.getLineMetrics(firstBeforeLoc.first_line);
+      var endBoundsBefore = dropletEditor.getLineMetrics(secondBeforeLoc.first_line);
+      startcoordsBefore = {pageX : startBoundsBefore.bounds.x, pageY: startBoundsBefore.bounds.y};
+      endcoordsBefore =  {pageX : endBoundsBefore.bounds.x, pageY: endBoundsBefore.bounds.y};
+      offset_top_before = startBoundsBefore.bounds.height - 30;
+      offset_left_before = Math.max(startBoundsBefore.bounds.width, endBoundsBefore.bounds.width)  + 20;
+
+    }
+    else{
+      offset_top_before = $(".editor").offset().top ;
+      offset_left_before = $(".editor").offset().left + 30;
+      startcoordsBefore = state.pane[pane].editor.renderer.textToScreenCoordinates((firstBeforeLoc.first_line), (firstBeforeLoc.last_column + 10));
+      endcoordsBefore = state.pane[pane].editor.renderer.textToScreenCoordinates((secondBeforeLoc.first_line ), (secondBeforeLoc.last_column + 10));
+    }
+    if (Math.abs(secondBeforeLoc.first_line - firstBeforeLoc.first_line) > 1){
+
+      var x_val = 0;
+      if(startcoordsBefore.pageX > endcoordsBefore.pageX){
+        x_val = startcoordsBefore.pageX;
+      } 
+      else{
+        x_val = endcoordsBefore.pageX;
+      }
+    }
+    if (show_fade){
```

Comment:
Here you are beginning to assemble some SVG.  It would be useful to have a comment illustrating an example of the SVG that you are creating.